### ISSUE TITLE
IndexError: tuple index out of range

### ISSUE THREAD

**Issue by wxz1989** [marin-m/pbtk#9]
when i parse a apk file, error in jar_extract.py show IndexError: tuple index out of range

`use_namer = all(len(i[4]) < 3 or i[4].endswith('_fld') for i in fields.values())
    
    all_vars = [field[4] for field in fields.values()]`

**Comment by marin-m** [comment]
Hello @wxz1989,

Thank you for your report. Can you please paste the last debugging lines that appear in your terminal before the error?

**Comment by wxz1989** [comment]
> Hello @wxz1989,
> 
> Thank you for your report. Can you please paste the last debugging lines that appear in your terminal before the error?

hi, this is the wrong log，thanks for your reply!

`
Traceback (most recent call last):
  File "/Users/xxx/Desktop/github_project/pbtk/gui.py", line 462, in run
    for name, contents in self.extractor['func'](input_):
  File "/Users/xxx/Desktop/github_project/pbtk/extractors/jar_extract.py", line 236, in handle_jar
    msg_path_to_obj, msg_to_referrers)
  File "/Users/xxx/Desktop/github_project/pbtk/extractors/jar_extract.py", line 680, in extract_lite
    all_vars = [field[4] for field in fields.values()]
  File "/Users/xxx/Desktop/github_project/pbtk/extractors/jar_extract.py", line 680, in <listcomp>
    all_vars = [field[4] for field in fields.values()]
IndexError: tuple index out of range`

**Comment by marin-m** [comment]
Hello,

Sorry, I rather meant, something like at least the ~40 last lines before the traceback for example. Don't hesitate to redact things if you don't want to disclose these. Don't hesitate to paste large logs output neither.

If you want to paste large logs, you can enclose these between backticks: ` ``` your text ``` ` (so that line breaks are preserved). It is also possible to past these on a site like https://gist.github.com/, and then share the links.

Regards,

**Comment by yoshimo** [comment]
Happens also with the dott app.
```
PS C:\pbtk> python .\gui.py

In com.google.protobuf.k:

In com.google.protobuf.t:

In com.google.protobuf.CodedOutputStream$c:

In com.google.firebase.firestore.l0.a:

In com.google.firebase.firestore.l0.c:

In com.google.firebase.firestore.l0.e:

In com.google.firebase.firestore.l0.g:

In com.google.firebase.firestore.l0.i:

In com.google.protobuf.d:

In com.google.protobuf.e0:

In com.google.protobuf.m:

In h.a.d.a.a:

In h.a.d.a.b0$c:

In h.a.d.a.b0$e:

In h.a.d.a.b0$h:

In h.a.d.a.b0$j:

In h.a.d.a.b0$l:

In h.a.d.a.b0$n:

In h.a.d.a.b0$p:

In h.a.d.a.b0$r:

In h.a.d.a.b0:

In h.a.d.a.c:

In h.a.d.a.d0$c:

In h.a.d.a.d0$e:

In h.a.d.a.d0:

In h.a.d.a.e0:

In h.a.d.a.e:

In h.a.d.a.f:

In h.a.d.a.h0:
{18: ('optional', 'bytes', None, 'java.lang.Double.valueOf(d1)', 'e'), 17: ('string', None), 11: ('optional', 'int32', None, 'java.lang.Double.valueOf(d1)', 'e'), 10: ('optional', 'message', 'com.google.protobuf.e0', 'java.lang.Double.valueOf(d1)', 'e'), 9: ('optional', 'message', 'h.a.d.a.a', 'java.lang.Double.valueOf(d1)', 'e'), 8: ('optional', 'message', 'h.a.g.a', 'java.lang.Double.valueOf(d1)', 'e'), 6: ('optional', 'message', 'h.a.d.a.x', 'java.lang.Double.valueOf(d1)', 'e'), 5: ('string', None), 3: ('optional', 'double', None, 'java.lang.Double.valueOf(d1)', 'e'), 2: ('optional', 'int64', None, 'java.lang.Double.valueOf(d1)', 'e'), 1: ('optional', 'bool', None, 'java.lang.Double.valueOf(d1)', 'e')}
Traceback (most recent call last):
  File ".\gui.py", line 462, in run
    for name, contents in self.extractor['func'](input_):
  File "C:\pbtk\extractors\jar_extract.py", line 236, in handle_jar
    msg_path_to_obj, msg_to_referrers)
  File "C:\pbtk\extractors\jar_extract.py", line 678, in extract_lite
    use_namer = all(len(i[4]) < 3 or i[4].endswith('_fld') for i in fields.values())
  File "C:\pbtk\extractors\jar_extract.py", line 678, in <genexpr>
    use_namer = all(len(i[4]) < 3 or i[4].endswith('_fld') for i in fields.values())
IndexError: tuple index out of range
```

**Comment by marin-m** [comment]
Thank you, should be fixed with commit a1bd560.
